powermon/commands/result.py

```diff
@@ -109,7 +109,15 @@ def decode_responses(self, responses=None) -> list[Reading]:
 
         # Process response based on result type
         match self.result_type:
-            case ResultType.ACK | ResultType.SINGLE | ResultType.MULTIVALUED:
+            case ResultType.ACK:
+                # Get the reading definition (there is only one)
+                reading_definition: ReadingDefinition = self.command.command_definition.get_reading_definition()
+                # Process the response using the reading_definition, into readings
+                readings = self.readings_from_response(responses, reading_definition)
+                if readings is not None:
```

Comment:
this seems valid for all 3 resulttypes - so i think this can be added to the original case and not split it

Reply:
So I had it at the beginning.
Then only thought to me, maybe divide so that you can do something with it.
I'll adapt it again.